// rxq/_esm5/Doc/index.js
var GetScriptEx = "GetScriptEx";
var SearchObjects = "SearchObjects";
var SearchResults = "SearchResults";
var SelectAssociations = "SelectAssociations";
var SearchAssociations = "SearchAssociations";
var SearchSuggest = "SearchSuggest";
var SendGenericCommandToCustomConnector = "SendGenericCommandToCustomConnector";
var GetFileTablesEx = "GetFileTablesEx";
var GetFileTablePreview = "GetFileTablePreview";
var GetFileTableFields = "GetFileTableFields";
var GetFileTables = "GetFileTables";
var GuessFileType = "GuessFileType";
var GetFolderItemsForConnection = "GetFolderItemsForConnection";
var GetDatabaseTablePreview = "GetDatabaseTablePreview";
var GetDatabaseTableFields = "GetDatabaseTableFields";
var GetDatabaseTables = "GetDatabaseTables";
var GetDatabaseOwners = "GetDatabaseOwners";
var GetDatabases = "GetDatabases";
var GetDatabaseInfo = "GetDatabaseInfo";
var GetConnections = "GetConnections";
var GetConnection = "GetConnection";
var DeleteConnection = "DeleteConnection";
var ModifyConnection = "ModifyConnection";
var CreateConnection = "CreateConnection";
var GetIncludeFileContent = "GetIncludeFileContent";
var SetFavoriteVariables = "SetFavoriteVariables";
var GetFavoriteVariables = "GetFavoriteVariables";
var CheckScriptSyntax = "CheckScriptSyntax";
var SetScript = "SetScript";
var GetSetAnalysis = "GetSetAnalysis";
var ForwardCount = "ForwardCount";
var BackCount = "BackCount";
var DoReloadEx = "DoReloadEx";
var GetLibraryContent = "GetLibraryContent";
var GetContentLibraries = "GetContentLibraries";
var GetMediaList = "GetMediaList";
var GetAssociationScores = "GetAssociationScores";
var SaveObjects = "SaveObjects";
var Scramble = "Scramble";
var FindMatchingFields = "FindMatchingFields";
var GetMatchingFields = "GetMatchingFields";
var Publish = "Publish";
var AbortModal = "AbortModal";
var Resume = "Resume";
var GetAllInfos = "GetAllInfos";
var GetFieldOnTheFlyByName = "GetFieldOnTheFlyByName";
var AddFieldFromExpression = "AddFieldFromExpression";
var CloneBookmark = "CloneBookmark";
var ApplyBookmark = "ApplyBookmark";
var GetBookmark = "GetBookmark";
var DestroyBookmark = "DestroyBookmark";
var CreateBookmark = "CreateBookmark";
var RemoveAlternateState = "RemoveAlternateState";
var AddAlternateState = "AddAlternateState";
var CheckNumberOrExpression = "CheckNumberOrExpression";
var CheckExpression = "CheckExpression";
var GetVariableByName = "GetVariableByName";
var GetVariableById = "GetVariableById";
var DestroyVariableByName = "DestroyVariableByName";
var DestroyVariableById = "DestroyVariableById";
var CreateVariableEx = "CreateVariableEx";
var DestroySessionVariable = "DestroySessionVariable";
var CreateSessionVariable = "CreateSessionVariable";
var CloneMeasure = "CloneMeasure";
var GetMeasure = "GetMeasure";
var DestroyMeasure = "DestroyMeasure";
var CreateMeasure = "CreateMeasure";
var CloneDimension = "CloneDimension";
var GetDimension = "GetDimension";
var DestroyDimension = "DestroyDimension";
var CreateDimension = "CreateDimension";
var ClearUndoBuffer = "ClearUndoBuffer";
var Redo = "Redo";
var Undo = "Undo";
var DestroyDraft = "DestroyDraft";
var CommitDraft = "CommitDraft";
var CreateDraft = "CreateDraft";
var CloneObject = "CloneObject";
var GetBookmarks = "GetBookmarks";
var GetObjects = "GetObjects";
var GetObject = "GetObject";
var DestroyObject = "DestroyObject";
var CreateObject = "CreateObject";
var DestroySessionObject = "DestroySessionObject";
var CreateSessionObject = "CreateSessionObject";
var GetLineage = "GetLineage";
var GetAppProperties = "GetAppProperties";
var SetAppProperties = "SetAppProperties";
var GetAppLayout = "GetAppLayout";
var GetTableData = "GetTableData";
var DoSave = "DoSave";
var SetFetchLimit = "SetFetchLimit";
var GetTextMacros = "GetTextMacros";
var GetScript = "GetScript";
var SetScriptBreakpoints = "SetScriptBreakpoints";
var GetScriptBreakpoints = "GetScriptBreakpoints";
var DoReload = "DoReload";
var GetEmptyScript = "GetEmptyScript";
var SetViewDlgSaveInfo = "SetViewDlgSaveInfo";
var GetViewDlgSaveInfo = "GetViewDlgSaveInfo";
var GetTablesAndKeys = "GetTablesAndKeys";
var GetLocaleInfo = "GetLocaleInfo";
var RemoveVariable = "RemoveVariable";
var CreateVariable = "CreateVariable";
var Forward = "Forward";
var Back = "Back";
var UnlockAll = "UnlockAll";
var LockAll = "LockAll";
var ClearAll = "ClearAll";
var EvaluateEx = "EvaluateEx";
var Evaluate = "Evaluate";
var SetLooselyCoupledVector = "SetLooselyCoupledVector";
var GetLooselyCoupledVector = "GetLooselyCoupledVector";
var GetVariable = "GetVariable";
var GetFieldDescription = "GetFieldDescription";
var GetField = "GetField";
export { GetField };
export { GetFieldDescription };
export { GetVariable };
export { GetLooselyCoupledVector };
export { SetLooselyCoupledVector };
export { Evaluate };
export { EvaluateEx };
export { ClearAll };
export { LockAll };
export { UnlockAll };
export { Back };
export { Forward };
export { CreateVariable };
export { RemoveVariable };
export { GetLocaleInfo };
export { GetTablesAndKeys };
export { GetViewDlgSaveInfo };
export { SetViewDlgSaveInfo };
export { GetEmptyScript };
export { DoReload };
export { GetScriptBreakpoints };
export { SetScriptBreakpoints };
export { GetScript };
export { GetTextMacros };
export { SetFetchLimit };
export { DoSave };
export { GetTableData };
export { GetAppLayout };
export { SetAppProperties };
export { GetAppProperties };
export { GetLineage };
export { CreateSessionObject };
export { DestroySessionObject };
export { CreateObject };
export { DestroyObject };
export { GetObject };
export { GetObjects };
export { GetBookmarks };
export { CloneObject };
export { CreateDraft };
export { CommitDraft };
export { DestroyDraft };
export { Undo };
export { Redo };
export { ClearUndoBuffer };
export { CreateDimension };
export { DestroyDimension };
export { GetDimension };
export { CloneDimension };
export { CreateMeasure };
export { DestroyMeasure };
export { GetMeasure };
export { CloneMeasure };
export { CreateSessionVariable };
export { DestroySessionVariable };
export { CreateVariableEx };
export { DestroyVariableById };
export { DestroyVariableByName };
export { GetVariableById };
export { GetVariableByName };
export { CheckExpression };
export { CheckNumberOrExpression };
export { AddAlternateState };
export { RemoveAlternateState };
export { CreateBookmark };
export { DestroyBookmark };
export { GetBookmark };
export { ApplyBookmark };
export { CloneBookmark };
export { AddFieldFromExpression };
export { GetFieldOnTheFlyByName };
export { GetAllInfos };
export { Resume };
export { AbortModal };
export { Publish };
export { GetMatchingFields };
export { FindMatchingFields };
export { Scramble };
export { SaveObjects };
export { GetAssociationScores };
export { GetMediaList };
export { GetContentLibraries };
export { GetLibraryContent };
export { DoReloadEx };
export { BackCount };
export { ForwardCount };
export { GetSetAnalysis };
export { SetScript };
export { CheckScriptSyntax };
export { GetFavoriteVariables };
export { SetFavoriteVariables };
export { GetIncludeFileContent };
export { CreateConnection };
export { ModifyConnection };
export { DeleteConnection };
export { GetConnection };
export { GetConnections };
export { GetDatabaseInfo };
export { GetDatabases };
export { GetDatabaseOwners };
export { GetDatabaseTables };
export { GetDatabaseTableFields };
export { GetDatabaseTablePreview };
export { GetFolderItemsForConnection };
export { GuessFileType };
export { GetFileTables };
export { GetFileTableFields };
export { GetFileTablePreview };
export { GetFileTablesEx };
export { SendGenericCommandToCustomConnector };
export { SearchSuggest };
export { SearchAssociations };
export { SelectAssociations };
export { SearchResults };
export { SearchObjects };
export { GetScriptEx };
