// rxq/_cjs/Doc/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var GetScriptEx = "GetScriptEx";
var SearchObjects = "SearchObjects";
var SearchResults = "SearchResults";
var SelectAssociations = "SelectAssociations";
var SearchAssociations = "SearchAssociations";
var SearchSuggest = "SearchSuggest";
var SendGenericCommandToCustomConnector = "SendGenericCommandToCustomConnector";
var GetFileTablesEx = "GetFileTablesEx";
var GetFileTablePreview = "GetFileTablePreview";
var GetFileTableFields = "GetFileTableFields";
var GetFileTables = "GetFileTables";
var GuessFileType = "GuessFileType";
var GetFolderItemsForConnection = "GetFolderItemsForConnection";
var GetDatabaseTablePreview = "GetDatabaseTablePreview";
var GetDatabaseTableFields = "GetDatabaseTableFields";
var GetDatabaseTables = "GetDatabaseTables";
var GetDatabaseOwners = "GetDatabaseOwners";
var GetDatabases = "GetDatabases";
var GetDatabaseInfo = "GetDatabaseInfo";
var GetConnections = "GetConnections";
var GetConnection = "GetConnection";
var DeleteConnection = "DeleteConnection";
var ModifyConnection = "ModifyConnection";
var CreateConnection = "CreateConnection";
var GetIncludeFileContent = "GetIncludeFileContent";
var SetFavoriteVariables = "SetFavoriteVariables";
var GetFavoriteVariables = "GetFavoriteVariables";
var CheckScriptSyntax = "CheckScriptSyntax";
var SetScript = "SetScript";
var GetSetAnalysis = "GetSetAnalysis";
var ForwardCount = "ForwardCount";
var BackCount = "BackCount";
var DoReloadEx = "DoReloadEx";
var GetLibraryContent = "GetLibraryContent";
var GetContentLibraries = "GetContentLibraries";
var GetMediaList = "GetMediaList";
var GetAssociationScores = "GetAssociationScores";
var SaveObjects = "SaveObjects";
var Scramble = "Scramble";
var FindMatchingFields = "FindMatchingFields";
var GetMatchingFields = "GetMatchingFields";
var Publish = "Publish";
var AbortModal = "AbortModal";
var Resume = "Resume";
var GetAllInfos = "GetAllInfos";
var GetFieldOnTheFlyByName = "GetFieldOnTheFlyByName";
var AddFieldFromExpression = "AddFieldFromExpression";
var CloneBookmark = "CloneBookmark";
var ApplyBookmark = "ApplyBookmark";
var GetBookmark = "GetBookmark";
var DestroyBookmark = "DestroyBookmark";
var CreateBookmark = "CreateBookmark";
var RemoveAlternateState = "RemoveAlternateState";
var AddAlternateState = "AddAlternateState";
var CheckNumberOrExpression = "CheckNumberOrExpression";
var CheckExpression = "CheckExpression";
var GetVariableByName = "GetVariableByName";
var GetVariableById = "GetVariableById";
var DestroyVariableByName = "DestroyVariableByName";
var DestroyVariableById = "DestroyVariableById";
var CreateVariableEx = "CreateVariableEx";
var DestroySessionVariable = "DestroySessionVariable";
var CreateSessionVariable = "CreateSessionVariable";
var CloneMeasure = "CloneMeasure";
var GetMeasure = "GetMeasure";
var DestroyMeasure = "DestroyMeasure";
var CreateMeasure = "CreateMeasure";
var CloneDimension = "CloneDimension";
var GetDimension = "GetDimension";
var DestroyDimension = "DestroyDimension";
var CreateDimension = "CreateDimension";
var ClearUndoBuffer = "ClearUndoBuffer";
var Redo = "Redo";
var Undo = "Undo";
var DestroyDraft = "DestroyDraft";
var CommitDraft = "CommitDraft";
var CreateDraft = "CreateDraft";
var CloneObject = "CloneObject";
var GetBookmarks = "GetBookmarks";
var GetObjects = "GetObjects";
var GetObject = "GetObject";
var DestroyObject = "DestroyObject";
var CreateObject = "CreateObject";
var DestroySessionObject = "DestroySessionObject";
var CreateSessionObject = "CreateSessionObject";
var GetLineage = "GetLineage";
var GetAppProperties = "GetAppProperties";
var SetAppProperties = "SetAppProperties";
var GetAppLayout = "GetAppLayout";
var GetTableData = "GetTableData";
var DoSave = "DoSave";
var SetFetchLimit = "SetFetchLimit";
var GetTextMacros = "GetTextMacros";
var GetScript = "GetScript";
var SetScriptBreakpoints = "SetScriptBreakpoints";
var GetScriptBreakpoints = "GetScriptBreakpoints";
var DoReload = "DoReload";
var GetEmptyScript = "GetEmptyScript";
var SetViewDlgSaveInfo = "SetViewDlgSaveInfo";
var GetViewDlgSaveInfo = "GetViewDlgSaveInfo";
var GetTablesAndKeys = "GetTablesAndKeys";
var GetLocaleInfo = "GetLocaleInfo";
var RemoveVariable = "RemoveVariable";
var CreateVariable = "CreateVariable";
var Forward = "Forward";
var Back = "Back";
var UnlockAll = "UnlockAll";
var LockAll = "LockAll";
var ClearAll = "ClearAll";
var EvaluateEx = "EvaluateEx";
var Evaluate = "Evaluate";
var SetLooselyCoupledVector = "SetLooselyCoupledVector";
var GetLooselyCoupledVector = "GetLooselyCoupledVector";
var GetVariable = "GetVariable";
var GetFieldDescription = "GetFieldDescription";
var GetField = "GetField";
exports.GetField = GetField;
exports.GetFieldDescription = GetFieldDescription;
exports.GetVariable = GetVariable;
exports.GetLooselyCoupledVector = GetLooselyCoupledVector;
exports.SetLooselyCoupledVector = SetLooselyCoupledVector;
exports.Evaluate = Evaluate;
exports.EvaluateEx = EvaluateEx;
exports.ClearAll = ClearAll;
exports.LockAll = LockAll;
exports.UnlockAll = UnlockAll;
exports.Back = Back;
exports.Forward = Forward;
exports.CreateVariable = CreateVariable;
exports.RemoveVariable = RemoveVariable;
exports.GetLocaleInfo = GetLocaleInfo;
exports.GetTablesAndKeys = GetTablesAndKeys;
exports.GetViewDlgSaveInfo = GetViewDlgSaveInfo;
exports.SetViewDlgSaveInfo = SetViewDlgSaveInfo;
exports.GetEmptyScript = GetEmptyScript;
exports.DoReload = DoReload;
exports.GetScriptBreakpoints = GetScriptBreakpoints;
exports.SetScriptBreakpoints = SetScriptBreakpoints;
exports.GetScript = GetScript;
exports.GetTextMacros = GetTextMacros;
exports.SetFetchLimit = SetFetchLimit;
exports.DoSave = DoSave;
exports.GetTableData = GetTableData;
exports.GetAppLayout = GetAppLayout;
exports.SetAppProperties = SetAppProperties;
exports.GetAppProperties = GetAppProperties;
exports.GetLineage = GetLineage;
exports.CreateSessionObject = CreateSessionObject;
exports.DestroySessionObject = DestroySessionObject;
exports.CreateObject = CreateObject;
exports.DestroyObject = DestroyObject;
exports.GetObject = GetObject;
exports.GetObjects = GetObjects;
exports.GetBookmarks = GetBookmarks;
exports.CloneObject = CloneObject;
exports.CreateDraft = CreateDraft;
exports.CommitDraft = CommitDraft;
exports.DestroyDraft = DestroyDraft;
exports.Undo = Undo;
exports.Redo = Redo;
exports.ClearUndoBuffer = ClearUndoBuffer;
exports.CreateDimension = CreateDimension;
exports.DestroyDimension = DestroyDimension;
exports.GetDimension = GetDimension;
exports.CloneDimension = CloneDimension;
exports.CreateMeasure = CreateMeasure;
exports.DestroyMeasure = DestroyMeasure;
exports.GetMeasure = GetMeasure;
exports.CloneMeasure = CloneMeasure;
exports.CreateSessionVariable = CreateSessionVariable;
exports.DestroySessionVariable = DestroySessionVariable;
exports.CreateVariableEx = CreateVariableEx;
exports.DestroyVariableById = DestroyVariableById;
exports.DestroyVariableByName = DestroyVariableByName;
exports.GetVariableById = GetVariableById;
exports.GetVariableByName = GetVariableByName;
exports.CheckExpression = CheckExpression;
exports.CheckNumberOrExpression = CheckNumberOrExpression;
exports.AddAlternateState = AddAlternateState;
exports.RemoveAlternateState = RemoveAlternateState;
exports.CreateBookmark = CreateBookmark;
exports.DestroyBookmark = DestroyBookmark;
exports.GetBookmark = GetBookmark;
exports.ApplyBookmark = ApplyBookmark;
exports.CloneBookmark = CloneBookmark;
exports.AddFieldFromExpression = AddFieldFromExpression;
exports.GetFieldOnTheFlyByName = GetFieldOnTheFlyByName;
exports.GetAllInfos = GetAllInfos;
exports.Resume = Resume;
exports.AbortModal = AbortModal;
exports.Publish = Publish;
exports.GetMatchingFields = GetMatchingFields;
exports.FindMatchingFields = FindMatchingFields;
exports.Scramble = Scramble;
exports.SaveObjects = SaveObjects;
exports.GetAssociationScores = GetAssociationScores;
exports.GetMediaList = GetMediaList;
exports.GetContentLibraries = GetContentLibraries;
exports.GetLibraryContent = GetLibraryContent;
exports.DoReloadEx = DoReloadEx;
exports.BackCount = BackCount;
exports.ForwardCount = ForwardCount;
exports.GetSetAnalysis = GetSetAnalysis;
exports.SetScript = SetScript;
exports.CheckScriptSyntax = CheckScriptSyntax;
exports.GetFavoriteVariables = GetFavoriteVariables;
exports.SetFavoriteVariables = SetFavoriteVariables;
exports.GetIncludeFileContent = GetIncludeFileContent;
exports.CreateConnection = CreateConnection;
exports.ModifyConnection = ModifyConnection;
exports.DeleteConnection = DeleteConnection;
exports.GetConnection = GetConnection;
exports.GetConnections = GetConnections;
exports.GetDatabaseInfo = GetDatabaseInfo;
exports.GetDatabases = GetDatabases;
exports.GetDatabaseOwners = GetDatabaseOwners;
exports.GetDatabaseTables = GetDatabaseTables;
exports.GetDatabaseTableFields = GetDatabaseTableFields;
exports.GetDatabaseTablePreview = GetDatabaseTablePreview;
exports.GetFolderItemsForConnection = GetFolderItemsForConnection;
exports.GuessFileType = GuessFileType;
exports.GetFileTables = GetFileTables;
exports.GetFileTableFields = GetFileTableFields;
exports.GetFileTablePreview = GetFileTablePreview;
exports.GetFileTablesEx = GetFileTablesEx;
exports.SendGenericCommandToCustomConnector = SendGenericCommandToCustomConnector;
exports.SearchSuggest = SearchSuggest;
exports.SearchAssociations = SearchAssociations;
exports.SelectAssociations = SelectAssociations;
exports.SearchResults = SearchResults;
exports.SearchObjects = SearchObjects;
exports.GetScriptEx = GetScriptEx;
