// rxq/_cjs/GenericObject/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var UnApprove = "UnApprove";
var Approve = "Approve";
var UnPublish = "UnPublish";
var Publish = "Publish";
var GetSnapshotObject = "GetSnapshotObject";
var EmbedSnapshotObject = "EmbedSnapshotObject";
var ResetMadeSelections = "ResetMadeSelections";
var EndSelections = "EndSelections";
var BeginSelections = "BeginSelections";
var CopyFrom = "CopyFrom";
var GetLinkedObjects = "GetLinkedObjects";
var SetChildArrayOrder = "SetChildArrayOrder";
var DestroyAllChildren = "DestroyAllChildren";
var DestroyChild = "DestroyChild";
var CreateChild = "CreateChild";
var GetChildInfos = "GetChildInfos";
var GetParent = "GetParent";
var GetChild = "GetChild";
var SelectHyperCubeContinuousRange = "SelectHyperCubeContinuousRange";
var MultiRangeSelectTreeDataValues = "MultiRangeSelectTreeDataValues";
var MultiRangeSelectHyperCubeValues = "MultiRangeSelectHyperCubeValues";
var RangeSelectHyperCubeValues = "RangeSelectHyperCubeValues";
var SelectPivotCells = "SelectPivotCells";
var SelectHyperCubeCells = "SelectHyperCubeCells";
var SelectHyperCubeValues = "SelectHyperCubeValues";
var Unlock = "Unlock";
var Lock = "Lock";
var DrillUp = "DrillUp";
var CollapseTop = "CollapseTop";
var CollapseLeft = "CollapseLeft";
var ExpandTop = "ExpandTop";
var ExpandLeft = "ExpandLeft";
var AcceptListObjectSearch = "AcceptListObjectSearch";
var AbortListObjectSearch = "AbortListObjectSearch";
var SearchListObjectFor = "SearchListObjectFor";
var SelectListObjectContinuousRange = "SelectListObjectContinuousRange";
var SelectListObjectAll = "SelectListObjectAll";
var SelectListObjectAlternative = "SelectListObjectAlternative";
var SelectListObjectExcluded = "SelectListObjectExcluded";
var SelectListObjectPossible = "SelectListObjectPossible";
var SelectListObjectValues = "SelectListObjectValues";
var ExportData = "ExportData";
var ClearSelections = "ClearSelections";
var GetInfo = "GetInfo";
var GetFullPropertyTree = "GetFullPropertyTree";
var SetFullPropertyTree = "SetFullPropertyTree";
var GetEffectiveProperties = "GetEffectiveProperties";
var GetProperties = "GetProperties";
var SetProperties = "SetProperties";
var ClearSoftPatches = "ClearSoftPatches";
var ApplyPatches = "ApplyPatches";
var GetHyperCubeBinnedData = "GetHyperCubeBinnedData";
var GetHyperCubeTreeData = "GetHyperCubeTreeData";
var GetHyperCubeContinuousData = "GetHyperCubeContinuousData";
var GetHyperCubeStackData = "GetHyperCubeStackData";
var GetHyperCubePivotData = "GetHyperCubePivotData";
var GetHyperCubeReducedData = "GetHyperCubeReducedData";
var GetHyperCubeData = "GetHyperCubeData";
var GetListObjectData = "GetListObjectData";
var GetLayout = "GetLayout";
exports.GetLayout = GetLayout;
exports.GetListObjectData = GetListObjectData;
exports.GetHyperCubeData = GetHyperCubeData;
exports.GetHyperCubeReducedData = GetHyperCubeReducedData;
exports.GetHyperCubePivotData = GetHyperCubePivotData;
exports.GetHyperCubeStackData = GetHyperCubeStackData;
exports.GetHyperCubeContinuousData = GetHyperCubeContinuousData;
exports.GetHyperCubeTreeData = GetHyperCubeTreeData;
exports.GetHyperCubeBinnedData = GetHyperCubeBinnedData;
exports.ApplyPatches = ApplyPatches;
exports.ClearSoftPatches = ClearSoftPatches;
exports.SetProperties = SetProperties;
exports.GetProperties = GetProperties;
exports.GetEffectiveProperties = GetEffectiveProperties;
exports.SetFullPropertyTree = SetFullPropertyTree;
exports.GetFullPropertyTree = GetFullPropertyTree;
exports.GetInfo = GetInfo;
exports.ClearSelections = ClearSelections;
exports.ExportData = ExportData;
exports.SelectListObjectValues = SelectListObjectValues;
exports.SelectListObjectPossible = SelectListObjectPossible;
exports.SelectListObjectExcluded = SelectListObjectExcluded;
exports.SelectListObjectAlternative = SelectListObjectAlternative;
exports.SelectListObjectAll = SelectListObjectAll;
exports.SelectListObjectContinuousRange = SelectListObjectContinuousRange;
exports.SearchListObjectFor = SearchListObjectFor;
exports.AbortListObjectSearch = AbortListObjectSearch;
exports.AcceptListObjectSearch = AcceptListObjectSearch;
exports.ExpandLeft = ExpandLeft;
exports.ExpandTop = ExpandTop;
exports.CollapseLeft = CollapseLeft;
exports.CollapseTop = CollapseTop;
exports.DrillUp = DrillUp;
exports.Lock = Lock;
exports.Unlock = Unlock;
exports.SelectHyperCubeValues = SelectHyperCubeValues;
exports.SelectHyperCubeCells = SelectHyperCubeCells;
exports.SelectPivotCells = SelectPivotCells;
exports.RangeSelectHyperCubeValues = RangeSelectHyperCubeValues;
exports.MultiRangeSelectHyperCubeValues = MultiRangeSelectHyperCubeValues;
exports.MultiRangeSelectTreeDataValues = MultiRangeSelectTreeDataValues;
exports.SelectHyperCubeContinuousRange = SelectHyperCubeContinuousRange;
exports.GetChild = GetChild;
exports.GetParent = GetParent;
exports.GetChildInfos = GetChildInfos;
exports.CreateChild = CreateChild;
exports.DestroyChild = DestroyChild;
exports.DestroyAllChildren = DestroyAllChildren;
exports.SetChildArrayOrder = SetChildArrayOrder;
exports.GetLinkedObjects = GetLinkedObjects;
exports.CopyFrom = CopyFrom;
exports.BeginSelections = BeginSelections;
exports.EndSelections = EndSelections;
exports.ResetMadeSelections = ResetMadeSelections;
exports.EmbedSnapshotObject = EmbedSnapshotObject;
exports.GetSnapshotObject = GetSnapshotObject;
exports.Publish = Publish;
exports.UnPublish = UnPublish;
exports.Approve = Approve;
exports.UnApprove = UnApprove;
