// rxq/_cjs/Global/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var GetBaseBNFString = "GetBaseBNFString";
var GetBaseBNFHash = "GetBaseBNFHash";
var GetBaseBNF = "GetBaseBNF";
var EngineVersion = "EngineVersion";
var GetStreamList = "GetStreamList";
var GetCustomConnectors = "GetCustomConnectors";
var GetSupportedCodePages = "GetSupportedCodePages";
var GetFolderItemsForPath = "GetFolderItemsForPath";
var GetLogicalDriveStrings = "GetLogicalDriveStrings";
var GetDefaultAppFolder = "GetDefaultAppFolder";
var IsValidConnectionString = "IsValidConnectionString";
var GetDatabasesFromConnectionString = "GetDatabasesFromConnectionString";
var GetOleDbProviders = "GetOleDbProviders";
var GetOdbcDsns = "GetOdbcDsns";
var GetFunctions = "GetFunctions";
var GetBNF = "GetBNF";
var CancelReload = "CancelReload";
var ConfigureReload = "ConfigureReload";
var GetAppEntry = "GetAppEntry";
var ProductVersion = "ProductVersion";
var CreateSessionAppFromApp = "CreateSessionAppFromApp";
var CreateSessionApp = "CreateSessionApp";
var OpenDoc = "OpenDoc";
var GetUniqueID = "GetUniqueID";
var IsPersonalMode = "IsPersonalMode";
var PublishApp = "PublishApp";
var ExportApp = "ExportApp";
var CopyApp = "CopyApp";
var ReplaceAppFromID = "ReplaceAppFromID";
var ReloadExtensionList = "ReloadExtensionList";
var ShutdownProcess = "ShutdownProcess";
var CancelRequest = "CancelRequest";
var IsDesktopMode = "IsDesktopMode";
var DeleteApp = "DeleteApp";
var CreateApp = "CreateApp";
var AllowCreateApp = "AllowCreateApp";
var GetActiveDoc = "GetActiveDoc";
var CreateDocEx = "CreateDocEx";
var GetAuthenticatedUser = "GetAuthenticatedUser";
var InteractDone = "InteractDone";
var GetInteract = "GetInteract";
var GetDocList = "GetDocList";
var QTProduct = "QTProduct";
var OSName = "OSName";
var OSVersion = "OSVersion";
var QvVersion = "QvVersion";
var GetProgress = "GetProgress";
var AbortAll = "AbortAll";
var AbortRequest = "AbortRequest";
exports.AbortRequest = AbortRequest;
exports.AbortAll = AbortAll;
exports.GetProgress = GetProgress;
exports.QvVersion = QvVersion;
exports.OSVersion = OSVersion;
exports.OSName = OSName;
exports.QTProduct = QTProduct;
exports.GetDocList = GetDocList;
exports.GetInteract = GetInteract;
exports.InteractDone = InteractDone;
exports.GetAuthenticatedUser = GetAuthenticatedUser;
exports.CreateDocEx = CreateDocEx;
exports.GetActiveDoc = GetActiveDoc;
exports.AllowCreateApp = AllowCreateApp;
exports.CreateApp = CreateApp;
exports.DeleteApp = DeleteApp;
exports.IsDesktopMode = IsDesktopMode;
exports.CancelRequest = CancelRequest;
exports.ShutdownProcess = ShutdownProcess;
exports.ReloadExtensionList = ReloadExtensionList;
exports.ReplaceAppFromID = ReplaceAppFromID;
exports.CopyApp = CopyApp;
exports.ExportApp = ExportApp;
exports.PublishApp = PublishApp;
exports.IsPersonalMode = IsPersonalMode;
exports.GetUniqueID = GetUniqueID;
exports.OpenDoc = OpenDoc;
exports.CreateSessionApp = CreateSessionApp;
exports.CreateSessionAppFromApp = CreateSessionAppFromApp;
exports.ProductVersion = ProductVersion;
exports.GetAppEntry = GetAppEntry;
exports.ConfigureReload = ConfigureReload;
exports.CancelReload = CancelReload;
exports.GetBNF = GetBNF;
exports.GetFunctions = GetFunctions;
exports.GetOdbcDsns = GetOdbcDsns;
exports.GetOleDbProviders = GetOleDbProviders;
exports.GetDatabasesFromConnectionString = GetDatabasesFromConnectionString;
exports.IsValidConnectionString = IsValidConnectionString;
exports.GetDefaultAppFolder = GetDefaultAppFolder;
exports.GetLogicalDriveStrings = GetLogicalDriveStrings;
exports.GetFolderItemsForPath = GetFolderItemsForPath;
exports.GetSupportedCodePages = GetSupportedCodePages;
exports.GetCustomConnectors = GetCustomConnectors;
exports.GetStreamList = GetStreamList;
exports.EngineVersion = EngineVersion;
exports.GetBaseBNF = GetBaseBNF;
exports.GetBaseBNFHash = GetBaseBNFHash;
exports.GetBaseBNFString = GetBaseBNFString;
