// rxq/_esm5/Global/index.js
var GetBaseBNFString = "GetBaseBNFString";
var GetBaseBNFHash = "GetBaseBNFHash";
var GetBaseBNF = "GetBaseBNF";
var EngineVersion = "EngineVersion";
var GetStreamList = "GetStreamList";
var GetCustomConnectors = "GetCustomConnectors";
var GetSupportedCodePages = "GetSupportedCodePages";
var GetFolderItemsForPath = "GetFolderItemsForPath";
var GetLogicalDriveStrings = "GetLogicalDriveStrings";
var GetDefaultAppFolder = "GetDefaultAppFolder";
var IsValidConnectionString = "IsValidConnectionString";
var GetDatabasesFromConnectionString = "GetDatabasesFromConnectionString";
var GetOleDbProviders = "GetOleDbProviders";
var GetOdbcDsns = "GetOdbcDsns";
var GetFunctions = "GetFunctions";
var GetBNF = "GetBNF";
var CancelReload = "CancelReload";
var ConfigureReload = "ConfigureReload";
var GetAppEntry = "GetAppEntry";
var ProductVersion = "ProductVersion";
var CreateSessionAppFromApp = "CreateSessionAppFromApp";
var CreateSessionApp = "CreateSessionApp";
var OpenDoc = "OpenDoc";
var GetUniqueID = "GetUniqueID";
var IsPersonalMode = "IsPersonalMode";
var PublishApp = "PublishApp";
var ExportApp = "ExportApp";
var CopyApp = "CopyApp";
var ReplaceAppFromID = "ReplaceAppFromID";
var ReloadExtensionList = "ReloadExtensionList";
var ShutdownProcess = "ShutdownProcess";
var CancelRequest = "CancelRequest";
var IsDesktopMode = "IsDesktopMode";
var DeleteApp = "DeleteApp";
var CreateApp = "CreateApp";
var AllowCreateApp = "AllowCreateApp";
var GetActiveDoc = "GetActiveDoc";
var CreateDocEx = "CreateDocEx";
var GetAuthenticatedUser = "GetAuthenticatedUser";
var InteractDone = "InteractDone";
var GetInteract = "GetInteract";
var GetDocList = "GetDocList";
var QTProduct = "QTProduct";
var OSName = "OSName";
var OSVersion = "OSVersion";
var QvVersion = "QvVersion";
var GetProgress = "GetProgress";
var AbortAll = "AbortAll";
var AbortRequest = "AbortRequest";
export { AbortRequest };
export { AbortAll };
export { GetProgress };
export { QvVersion };
export { OSVersion };
export { OSName };
export { QTProduct };
export { GetDocList };
export { GetInteract };
export { InteractDone };
export { GetAuthenticatedUser };
export { CreateDocEx };
export { GetActiveDoc };
export { AllowCreateApp };
export { CreateApp };
export { DeleteApp };
export { IsDesktopMode };
export { CancelRequest };
export { ShutdownProcess };
export { ReloadExtensionList };
export { ReplaceAppFromID };
export { CopyApp };
export { ExportApp };
export { PublishApp };
export { IsPersonalMode };
export { GetUniqueID };
export { OpenDoc };
export { CreateSessionApp };
export { CreateSessionAppFromApp };
export { ProductVersion };
export { GetAppEntry };
export { ConfigureReload };
export { CancelReload };
export { GetBNF };
export { GetFunctions };
export { GetOdbcDsns };
export { GetOleDbProviders };
export { GetDatabasesFromConnectionString };
export { IsValidConnectionString };
export { GetDefaultAppFolder };
export { GetLogicalDriveStrings };
export { GetFolderItemsForPath };
export { GetSupportedCodePages };
export { GetCustomConnectors };
export { GetStreamList };
export { EngineVersion };
export { GetBaseBNF };
export { GetBaseBNFHash };
export { GetBaseBNFString };
